main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+/*
+example:
+	*{"name": "Ilya", "hobby": "Programming"}
+	*localhost:9000/?id=1
+*/
+
+// User represents user with name and hobby strings fields.
+type User struct {
+	Name  string `json:"name"`
+	Hobby string `json:"hobby"`
+}
+
+var idUserDB = make(map[int]User)
+
+var nameHobbyDB = make(map[string]string)
+
+var number int
+
+func main() {
+	http.HandleFunc("/", handler)
+	err := http.ListenAndServe("localhost:9000", nil)
+	if err != nil {
+		log.Fatal("failed to start server", err)
+	}
+}
+
+func handler(writer http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case http.MethodGet:
+		getUser(writer, req)
+	case http.MethodPost:
+		addUser(writer, req)
+	}
+}
+
+func addUser(writer http.ResponseWriter, req *http.Request) {
+	var user User
+	err := json.NewDecoder(req.Body).Decode(&user)
+	if err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		fmt.Fprintf(writer, "cannot decode json: %s", err)
```

Comment:
I think you should user `\n` here as well, or remove `\n` from everywhere below. Let's try to be consistent